### PR TITLE
update ci

### PR DESCRIPTION
**Pull request by anishamidi**

### CODE REVIEW

**Comment by 1995parham** [comment]
Thanks, Anis jan, we can also use new docker build which uses bake.

File: .github/workflows/build-release.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
+---
+  name: build-release
+  on:
+    push:
+      tags: [ v* ]
+  
+  jobs:
+    lint:
```
Comment (hoptical) [inline .github/workflows/build-release.yml]: Lint job must not be in the `build-release` workflow. It should be in a separate one triggered with each commit.

File: .github/workflows/build-release.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        - uses: actions/setup-go@v4
+          with:
+            go-version: '1.21'
+            cache: false
+        - name: golangci-lint
+          uses: golangci/golangci-lint-action@v3
+          with:
+            version: latest
+            args: --timeout 5m
+    test:
```
Comment (hoptical) [inline .github/workflows/build-release.yml]: Same as lint. Not suitable here.

File: .github/workflows/build-release.yml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      steps:
+        - uses: actions/checkout@v3
+        - name: run tests using make
+          run: make test
+    docker:
+      name: docker
+      runs-on: ubuntu-latest
+      needs:
+        - lint
+        - test
```
Comment (hoptical) [inline .github/workflows/build-release.yml]: Not needed.

File: .github/workflows/build-release.yml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      steps:
+        - uses: actions/checkout@v2
+        - uses: docker/setup-qemu-action@v1
+        - uses: docker/setup-buildx-action@v1
+        - uses: docker/login-action@v1
+          with:
+            registry: ghcr.io
+            username: ${{ github.repository_owner }}
+            password: ${{ secrets.GITHUB_TOKEN }}
+        - uses: docker/metadata-action@v3
```
Comment (hoptical) [inline .github/workflows/build-release.yml]: Use `docker/metadata-action@v4`

File: .github/workflows/build-release.yml [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+      name: docker
+      runs-on: ubuntu-latest
+      needs:
+        - lint
+        - test
+      steps:
+        - uses: actions/checkout@v2
+        - uses: docker/setup-qemu-action@v1
+        - uses: docker/setup-buildx-action@v1
+        - uses: docker/login-action@v1
```
Comment (hoptical) [inline .github/workflows/build-release.yml]: use v2 instead of v1

File: .github/workflows/build-release.yml [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    helm-chart:
+      name: helm-chart
+      needs:
+          - docker
+      runs-on: ubuntu-latest
+      steps:
+        - name: Checkout
+          uses: actions/checkout@v3
+          with:
+            fetch-depth: 0
```
Comment (hoptical) [inline .github/workflows/build-release.yml]: Not needed

File: .github/workflows/build-release.yml [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+  
+        - name: Configure Git
```
Comment (hoptical) [inline .github/workflows/build-release.yml]: Please replace with the s3-operator one: https://github.com/snapp-incubator/s3-operator/blob/main/.github/workflows/build-release.yaml#L56-L67

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -1,66 +1,65 @@
...
-        with:
-          file: "Dockerfile"
-          context: .
-          platforms: linux/amd64
-          push: true
-          tags: |
-            ${{ steps.meta.outputs.tags }}
-          labels: ${{ steps.meta.outputs.labels }}
-
+  name: ci
```
Comment (hoptical) [inline .github/workflows/ci.yml]: Bad name. Suggestion: `checks`.

File: .github/workflows/ci.yml [hunk 9]
```diff
@@ -1,66 +1,65 @@
...
+  on:
+    push:
+      branches: [ main ]
```
Comment (hoptical) [inline .github/workflows/ci.yml]: add PR trigger, too:
```yaml
  pull_request:
```
example: https://github.com/snapp-incubator/s3-operator/blob/main/.github/workflows/checks.yaml#L3-L7

File: .github/workflows/ci.yml [hunk 10]
```diff
@@ -1,66 +1,65 @@
...
+  
+        - uses: docker/build-push-action@v2
+          with:
+            file: "Dockerfile"
+            context: .
+            platforms: linux/amd64
+            push: true
+            tags: |
+              ${{ steps.meta.outputs.tags }}
+            labels: ${{ steps.meta.outputs.labels }}
```
Comment (hoptical) [inline .github/workflows/ci.yml]: Not needed.

File: .github/workflows/release.yaml
Comment (hoptical) [inline .github/workflows/release.yaml:1]: This file must be in the parent directory (i.e. `.github` folder) and its contents can be as [this](https://github.com/snapp-incubator/s3-operator/blob/main/.github/release.yaml)

**Review by hoptical — APPROVED** [review]
Thanks,  @anishamidi  jan.
LGTM.

**Pull request merged**